crates/sui-snapshot/src/reader.rs

```diff
@@ -376,54 +400,49 @@ impl StateSnapshotReaderV1 {
                                 .get(part_num)
                                 .expect("sha3 digest not in bucket map");
 
-                            Ok::<(Bytes, FileMetadata, [u8; 32]), anyhow::Error>((
-                                bytes,
-                                (*file_metadata).clone(),
-                                *sha3_digest,
-                            ))
+                            tokio::spawn({
```

Comment:
should this be spawn blocking?

Reply:
it was a pain to get this to work due to lifetime issues it creates. Given that this doesn't seem a bottleneck, could we revisit making this `spawn_blocking` alongside the other optimization followups?